buildSrc/src/main/java/org/elasticsearch/test/NamingConventionsCheck.java

```diff
@@ -63,59 +59,61 @@ public static void main(String[] args) throws IOException {
-                assertViolation("InnerTests", check.innerClasses);
-                assertViolation("NotImplementingTests", check.notImplementing);
-                assertViolation("PlainUnit", check.pureUnitTest);
+                check.checkTests(rootPath, skipIntegTestsInDisguise);
             }
         }
 
         // Now we should have no violations
-        assertNoViolations(
+        int exitCode  = 0 ;
+        exitCode += countAndPrintViolations(
                 "Not all subclasses of " + check.testClass.getSimpleName()
                     + " match the naming convention. Concrete classes must end with [Tests]",
-                check.missingSuffix);
-        assertNoViolations("Classes ending with [Tests] are abstract or interfaces", check.notRunnable);
-        assertNoViolations("Found inner classes that are tests, which are excluded from the test runner", check.innerClasses);
-        assertNoViolations("Pure Unit-Test found must subclass [" + check.testClass.getSimpleName() + "]", check.pureUnitTest);
-        assertNoViolations("Classes ending with [Tests] must subclass [" + check.testClass.getSimpleName() + "]", check.notImplementing);
-        assertNoViolations(
-                "Classes ending with [Tests] or [IT] or extending [" + check.testClass.getSimpleName() + "] must be in src/test/java",
-                check.testsInMain);
+                check.missingSuffix) ;
+        exitCode += countAndPrintViolations(
+            "Classes ending with [Tests] are abstract or interfaces",
+            check.notRunnable
+        );
+        exitCode += countAndPrintViolations(
+            "Found inner classes that are tests, which are excluded from the test runner",
+            check.innerClasses
+        );
+        exitCode += countAndPrintViolations(
+            "Pure Unit-Test found must subclass [" + check.testClass.getSimpleName() + "]",
+            check.pureUnitTest
+        );
+        exitCode += countAndPrintViolations(
+            "Classes ending with [Tests] must subclass [" + check.testClass.getSimpleName() + "]",
+            check.notImplementing
+        );
+        exitCode += countAndPrintViolations(
+                "Classes ending with [Tests] or [IT] or extending [" +
+                    check.testClass.getSimpleName() + "] must be in src/test/java",
+                check.testsInMain
+        );
         if (skipIntegTestsInDisguise == false) {
-            assertNoViolations(
-                    "Subclasses of " + check.integTestClass.getSimpleName() + " should end with IT as they are integration tests",
-                    check.integTestsInDisguise);
+            exitCode += countAndPrintViolations("Subclasses of " + check.integTestClass.getSimpleName() +
+                        " should end with IT as they are integration tests",
+                    check.integTestsInDisguise
+                );
         }
+        System.exit(exitCode);
```

Comment:
I don't think we should call System.exit. Instead, throw a GradleException with a message indicating there were violations.

Reply:
This uses an exit code since we don't call into this code directly, but run it like a cli utility.